### PR TITLE
47 create account username new

### LINKED ISSUE #47: add username to firestore on create account
implement check if username has been taken

### PR DESCRIPTION
**Pull request by tamari-gray**
think fixes #47 

shes a big one 🙏

### CODE REVIEW

**Review by nickmeinhold — DISMISSED** [review]
Looks awesome mate!!! 

I didn't realise how many of the files were just the migration... it was actually a pretty quick review!  

Onwards and upwards! 👍

**Comment by nickmeinhold** [comment]
Oh I missed the failing test, my bad. I will leave for you to merge @tamari-gray when the test is fixed. Let me know if I can help! :-)

File: test/auth_section/create_account/widget_test.dart [hunk 1]
```diff
@@ -4,19 +4,23 @@ import 'package:mockito/mockito.dart';
...
 
 class MockNavService extends Mock implements NavigationService {}
 
+class MockDBService extends Mock implements DatabaseService {}
+
 void main() {
-  Widget makeTestableCreateAccountWidget(
-      MockNavService mockNavService, FakeAuthService mockAuth) {
+  Widget makeTestableCreateAccountWidget(MockNavService mockNavService,
+      FakeAuthService mockAuth, MockDBService mockDBService) {
```
Comment (nickmeinhold) [inline test/auth_section/create_account/widget_test.dart:19]: Should `mockAuth` be `fakeAuth`?

File: test/auth_section/create_account/widget_test.dart [hunk 2]
```diff
@@ -30,8 +34,9 @@ void main() {
       //set up for testing
       final _mockNavService = MockNavService();
       final _mockAuthService = FakeAuthService(_mockNavService, true);
-      await tester.pumpWidget(
-          makeTestableCreateAccountWidget(_mockNavService, _mockAuthService));
+      final _mockDBService = MockDBService();
+      await tester.pumpWidget(makeTestableCreateAccountWidget(
+          _mockNavService, _mockAuthService, _mockDBService));
```
Comment (nickmeinhold) [inline test/auth_section/create_account/widget_test.dart:39]: here you've gone with no trailing comma while below you used the trailing comma, so dartfmt puts each parameter on it's own line - definitely a nitpick but I reckon if you choose one style (trailing comma or no trailing comma) and stick with it your code will be more readable

File: test/auth_section/create_account/widget_test.dart [hunk 3]
```diff
@@ -67,8 +72,13 @@ void main() {
       final _mockNavService = MockNavService();
       // return firebase auth error
       final _mockAuthService = FakeAuthService(_mockNavService, false);
-      await tester.pumpWidget(
-          makeTestableCreateAccountWidget(_mockNavService, _mockAuthService));
+      final _mockDBService = MockDBService();
+
+      await tester.pumpWidget(makeTestableCreateAccountWidget(
```
Comment (nickmeinhold) [inline test/auth_section/create_account/widget_test.dart:77]: this is an example of the trailing comma I was talking about in the previous comment

**Review by nickmeinhold — DISMISSED** [review]
Looks great man!  Awesome work!!!

**Comment by nickmeinhold** [comment]
@tamari-gray I'm leaving for you to merge again so you can decide if you want to change anything based on the comments or not.  Should we default to this - ie. reviewer approves and leaves the author to merge?  Cheers mate!

**Review by nickmeinhold — APPROVED** [review]
👍

File: test/auth_section/create_account/widget_test.dart (hunk 1 above)
Comment (tamari-gray) [reply test/auth_section/create_account/widget_test.dart:19]: yeah :)

**Pull request merged**